hibernate-core/src/main/java/org/hibernate/jpa/boot/internal/PersistenceXmlParser.java

```diff
@@ -62,7 +64,7 @@
 	 */
 	public static List<ParsedPersistenceXmlDescriptor> locatePersistenceUnits(Map integration) {
 		final PersistenceXmlParser parser = new PersistenceXmlParser(
```

Comment:
This should be changed quite a bit IMO....

`#locatePersistenceUnits` is only ever called from one place in "real code".  That call already has access to the `ClassLoaderService`.  We should just pass that one along.  The other uses are tests.  For those we can just create the CLS, or get it from the `ServiceRegistry` if they have access to one.

`#locateIndividualPersistenceUnit` - None of the overloads are ever called from outside this class.  IMO we should just remove them.  This is an internal class, so changing stuff is kosher.

`#locateNamedPersistenceUnit` - Same as `#locateIndividualPersistenceUnit`

`#parse` - One form is never used.  The other is used from a context that does not (atm) have access to CLS.  So we would need to do something there.  Though I think we want to find an option here that does not involve creating 2 CLS refs.  This method is called when creating an `EntityManagerFactoryBuilder`.  That EMFB will eventually create a CLS (or use one passed in).  So not only is it something I'd prefer to avoid perf wise[1], but it is potentially wrong if we are passed a CLS or ClassLoader to build into a CLS.


[1] Yes this "throw away" CLS is only created and used at boot time.  Still...